tests/src/Kernel/fixtures/rendering.yml

```diff
@@ -2950,3 +2950,189 @@
+      div.ecl-list-illustration__image[style="background-image:url('https://placeimg.com/1000/500/arch')"]: 1
+      div.ecl-list-illustration__image[aria-label="Alternative text for featured item image 2"]: 1
+      div.ecl-list-illustration__image[style="background-image:url('https://placeimg.com/500/500/arch')"]: 1
+    contains:
+      'div.ecl-list-illustration__item:nth-child(1) div.ecl-list-illustration__title': 'Business, Economy, Euro'
+      'div.ecl-list-illustration__item:nth-child(1) div.ecl-list-illustration__description': 'EU economy, the euro, and practical information for EU businesses and entrepreneurs.'
+      'div.ecl-list-illustration__item:nth-child(2) div.ecl-list-illustration__title': 'About the European Union'
+      'div.ecl-list-illustration__item:nth-child(2) div.ecl-list-illustration__description': 'The EU and its institutions, how to visit and work at the EU.'
+- array:
+    '#type': 'pattern'
+    '#id': 'list_with_illustration'
+    '#variant': "horizontal_images_square"
+    '#fields':
+      column: 3
+      items:
+        - title: "Business, Economy, Euro"
+          description: "EU economy, the euro, and practical information for EU businesses and entrepreneurs."
+          image:
+            src: "https://placeimg.com/1000/500/arch"
+            alt: "Alternative text for featured item image 1"
+        - title: "About the European Union"
+          description: "The EU and its institutions, how to visit and work at the EU."
+          image:
+            src: "https://placeimg.com/500/500/arch"
+            alt: "Alternative text for featured item image 2"
+  assertions:
+    count:
+      div.ecl-list-illustration: 1
+      div.ecl-list-illustration.ecl-list-illustration--col-3: 1
+      div.ecl-list-illustration__item: 2
+      div.ecl-list-illustration__image: 2
+      div.ecl-list-illustration__image.ecl-list-illustration__image--square: 2
+      div.ecl-list-illustration__image[aria-label="Alternative text for featured item image 1"]: 1
+      div.ecl-list-illustration__image[style="background-image:url('https://placeimg.com/1000/500/arch')"]: 1
+      div.ecl-list-illustration__image[aria-label="Alternative text for featured item image 2"]: 1
+      div.ecl-list-illustration__image[style="background-image:url('https://placeimg.com/500/500/arch')"]: 1
+    contains:
+      'div.ecl-list-illustration__item:nth-child(1) div.ecl-list-illustration__title': 'Business, Economy, Euro'
+      'div.ecl-list-illustration__item:nth-child(1) div.ecl-list-illustration__description': 'EU economy, the euro, and practical information for EU businesses and entrepreneurs.'
+      'div.ecl-list-illustration__item:nth-child(2) div.ecl-list-illustration__title': 'About the European Union'
+      'div.ecl-list-illustration__item:nth-child(2) div.ecl-list-illustration__description': 'The EU and its institutions, how to visit and work at the EU.'
+- array:
```

Comment:
Can we add test coverage for zebra enabled on horizontal to see that it gets disabled? 
Also for uses cases when the variant is one of the image variants containing items with icons and vice versa to ensure the template adapts the final items array to the ECL component.